tests/utils/test_multiprocessing.py

```
@@ -306,3 +307,40 @@ def test_stop_runs_housekeeping_after_interrupt():
         unreg.assert_called_once_with(MPManager.stop)
     assert _ManagerSingleton.has_cls(MPManager) is False
+
+
+@pytest.mark.parametrize("func", ["fork", "forkpty"])
+def test_forking_suppresses_fork_thread_warning(func):
+    message = f"This process (pid=1) is multi-threaded, use of {func}() may " \
+              "lead to deadlocks in the child."
+    with warnings.catch_warnings(record=True) as caught:
+        warnings.simplefilter("always")
+        with forking():
+            warnings.warn(message, DeprecationWarning)
```

Ruff (0.15.21)

[warning] No explicit `stacklevel` keyword argument found

Set `stacklevel=2`

(B028)